src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -792,8 +793,17 @@ private BoundLambda ReallyBindForErrorRecovery()
 
             return
                 GuessBestBoundLambda(_bindingCache)
-                ?? GuessBestBoundLambda(_returnInferenceCache)
+                ?? rebind(GuessBestBoundLambda(_returnInferenceCache))
                 ?? ReallyInferReturnType(null, ImmutableArray<TypeWithAnnotations>.Empty, ImmutableArray<RefKind>.Empty);
+
+            // Rebind a lambda to push target conversions through the return/result expressions
+            BoundLambda rebind(BoundLambda lambda)
```

Comment:
>BoundLambda rebind(BoundLambda lambda) [](start = 12, length = 38)

Consider making this local function static by adding necessary pieces as explicit parameters.

Reply:
I prefer not to as it uses `this`, so acts more like an instance method.

---
In reply to: [328269540](https://github.com/dotnet/roslyn/pull/38750#discussion_r328269540) [](ancestors = 328269540)